src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -1344,9 +1354,48 @@ private static bool IsLessDerivedThanAny<TMember>(int index, TypeSymbol type, Ar
                 {
                     return true;
                 }
-            }
+                else if (currentType.IsExtension)
+                {
+                    if (currentType.ExtendedTypeNoUseSiteDiagnostics is { } extendedType)
```

Comment:
Consider merging these two `if`s into one.
```suggestion
                else if (currentType.IsExtension && currentType.ExtendedTypeNoUseSiteDiagnostics is { } extendedType)
```

Reply:
The reason I kept those two separate is that it's possible to have an extension type with a `null` `ExtendedTypeNoUseSiteDiagnostics` in some error scenarios. With the current structure, if we have an extension, we won't at a subsequent `else if` below if one gets added in the future, regardless of the value found in `ExtendedTypeNoUseSiteDiagnostics`.